subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/FixedSizeOldestCacheCleanup.java

```diff
@@ -135,38 +70,7 @@ public void run(BuildOperationContext context) {
         return filesForDeletion;
     }
 
-    File[] findEligibleFiles(File cacheDir) {
-        // TODO: This doesn't descend subdirectories.
-        return cacheDir.listFiles(new FilenameFilter() {
-            @Override
-            public boolean accept(File dir, String name) {
-                return canBeDeleted(name);
-            }
-        });
-    }
-
-    void cleanupFiles(final PersistentCache persistentCache, final List<File> filesForDeletion) {
-        // Need to remove some files
-        long removedSize = deleteFiles(filesForDeletion);
-        LOGGER.info("{} removing {} cache entries ({} reclaimed).", persistentCache, filesForDeletion.size(), FileUtils.byteCountToDisplaySize(removedSize));
-    }
-
-    private long deleteFiles(List<File> files) {
-        long removedSize = 0;
-        for (File file : files) {
-            try {
-                long size = file.length();
-                if (file.delete()) {
-                    removedSize += size;
-                }
-            } catch (Exception e) {
-                LOGGER.debug("Could not clean up cache " + file, e);
-            }
-        }
-        return removedSize;
-    }
-
-    boolean canBeDeleted(String name) {
-        return !(name.endsWith(".properties") || name.endsWith(".lock") || name.endsWith(partialFileSuffix));
+    protected boolean canBeDeleted(String name) {
```

Comment:
I wouldn't make this part of the contract with the generic cleaning stuff. This action can do its own filtering in the 'select some files to delete' method.

Reply:
I raised this with Lorant/Stefan and I think we don't need to do this now (the `.part` stuff was for troubleshooting).